src/helpers/validation.ts

```diff
@@ -0,0 +1,21 @@
+import { Request,Response,NextFunction } from "express";
+export const dataValidation = async(req:Request,res:Response,next:NextFunction,data:any) => {
+    const {error} = data.validate(req.body);
+    if(error){
+       return res.status(406)
+        .json({
+            status:406,
+            mesage:`Error in User Data : ${error.message}`
+        })
+    }
+    const allowedFields = Object.keys(data.describe().keys);
+    const unknownFields = Object.keys(req.body).filter(field => !allowedFields.includes(field));
+    if (unknownFields.length > 0) {
+        return res.status(406).json({
```

Comment:
wrong status u should use 400 here 406 is used for unacceptable which is related to content type but u used it for field validations